scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -2053,6 +2053,49 @@ def scc_order(self) -> List[List[Node]]:
         except Exception as e:
             raise EDTError(e)
 
+    def _link_properties(self, parent_node, node):
+        """
+        Link node properties to parent_node. Iterates over child nodes
+        recursively if necessary.
+        """
+
+        # A Node depends on any Nodes present in 'phandle',
+        # 'phandles', or 'phandle-array' property values.
+        for prop in node.props.values():
+            if prop.type == 'phandle':
+                self._graph.add_edge(parent_node, prop.val)
+            elif prop.type == 'phandles':
+                if TYPE_CHECKING:
+                    assert isinstance(prop.val, list)
+                for phandle_node in prop.val:
+                    self._graph.add_edge(parent_node, phandle_node)
+            elif prop.type == 'phandle-array':
+                if TYPE_CHECKING:
+                    assert isinstance(prop.val, list)
+                for cd in prop.val:
+                    if cd is None:
+                        continue
+                    if TYPE_CHECKING:
+                        assert isinstance(cd, ControllerAndData)
+                    self._graph.add_edge(parent_node, cd.controller)
+
+        # A Node depends on whatever supports the interrupts it
+        # generates.
+        for intr in node.interrupts:
+            self._graph.add_edge(parent_node, intr.controller)
+
+        # If the binding defines child bindings, link the child properties to
+        # the original node as well.
+        if node._binding and node._binding.child_binding:
+            for child in node.children.values():
+                if "compatible" in child.props:
+                    continue
+                if node._binding.path != child._binding.path:
+                    # Not a child binding, normal child node on a different
```

Comment:
Why is this separate check needed; doesn't the compatible check on line 2091 cover this?

Reply:
Yeah good point, wrote this long time ago and now that you point it out it does indeed seem like 2091 alone is enough. I dropped it and gave it a push CI run as well, everything seems to check out so let's drop it. Thanks for spotting this.